base_layer/transaction_components/src/key_manager/inner.rs

```diff
@@ -1592,6 +1678,22 @@ where TBackend: TransactionKeyManagerBackend + 'static
         Ok(CompressedPublicKey::new_from_pk(public_key))
     }
 
+    pub async fn stealth_address_script_spending_key_id(
+        &self,
+        commitment_mask_key_id: &TariKeyId,
+        spend_key_id: &TariKeyId,
+    ) -> Result<PrivateKey, TransactionError> {
```

Comment:
```suggestion
    pub async fn stealth_address_script_spending_key_id(
        &self,
        commitment_mask_key_id: &TariKeyId,
        spend_key_id: &TariKeyId,
    ) -> Result<TariKeyId, TransactionError> {
```
This method should import the key and not resturn the private key